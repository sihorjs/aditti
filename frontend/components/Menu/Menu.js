import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { postSignOut } from "actions/auth";
import "./Menu.less";

function Menu(props) {
    return props.menu.length ? (
        <nav className="main-menu">
            <ul className="main-menu__container wrapper">
                {props.menu.map(item => {
                    return (
                        <li className="main-menu__item" key={item.name}>
                            <Link
                                className="main-menu__link"
                                to={item.href}
                            >
                                {item.name}
                            </Link>
                        </li>
                    );
                })}
                {props.isAuthenticated ? (
                    <li className="main-menu__item">
                        <button
                            className="main-menu__link"
                            onClick={props.onSignOut}
                            type="button"
                        >
                            Sign out
                        </button>
                    </li>
                ) : (
                    <Fragment>
                        <li className="main-menu__item">
                            <Link
                                className="main-menu__link"
                                to="/signup"
                            >
                                Sign up
                            </Link>
                        </li>
                        <li className="main-menu__item">
                            <Link
                                className="main-menu__link"
                                to="/signin"
                            >
                                Sign in
                            </Link>
                        </li>
                    </Fragment>
                )}
            </ul>
        </nav>
    ) : null;
}

Menu.propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    menu: PropTypes.arrayOf(
        PropTypes.shape({
            name: PropTypes.string.isRequired,
            href: PropTypes.string.isRequired
        })
    ).isRequired,
    onSignOut: PropTypes.func.isRequired
};

function mapStateToProps(state) {
    return {
        isAuthenticated: state.auth.status
    };
}

function mapDispatchToProps(dispatch) {
    return {
        onSignOut: () => dispatch(postSignOut())
    };
}

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Menu);
