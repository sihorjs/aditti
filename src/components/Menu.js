import React from "react";
import PropTypes from "prop-types";

function Menu(props) {
    return (
        <nav id="main-menu">
            <ul>
                {props.menu.map((item, i) => {
                    return (
                        <li key={i}>
                            <a href={item.href}>{item.name}</a>
                        </li>
                    );
                })}
            </ul>
        </nav>
    );
}

Menu.propTypes = {
    menu: PropTypes.arrayOf({
        name: PropTypes.string.isRequired,
        href: PropTypes.string.isRequired
    }).isRequired
};

export default Menu;
