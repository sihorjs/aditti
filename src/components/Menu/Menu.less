@import url("../../less/index.less");

.main-menu {
    border: solid 1px @gray;
    background-color: @cyan;

    &__container {
        display: flex;
        width: @desktop-width;
        margin: 0 auto;
        text-align: center;
        font: 21px/35px @heading-font;
        color: @white;
    }

    &__item {
        position: relative;

        &:not(:last-child)::after {
            position: absolute;
            content: "";
            top: 50%;
            left: 100%;
            margin-left: -1px;
            z-index: 1;
            width: 2px;
            height: 50%;
            background-color: @white;
            transform: translate(50%, 0);
        }
    }

    &__link {
        display: block;
        padding: 0 15px;
        color: @white;

        &:hover {
            background-color: @darkcyan;
        }
    }
}
