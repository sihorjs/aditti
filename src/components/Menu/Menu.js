import React from "react";
import PropTypes from "prop-types";

import "./Menu.less";

function Menu(props) {
    return (
        <nav className="main-menu">
            <ul className="main-menu__container">
                {props.menu.map(item => {
                    return (
                        <li key={item.name} className="main-menu__item">
                            <a className="main-menu__link" href={item.href}>
                                {item.name}
                            </a>
                        </li>
                    );
                })}
            </ul>
        </nav>
    );
}

Menu.propTypes = {
    menu: PropTypes.arrayOf(
        PropTypes.shape({
            name: PropTypes.string.isRequired,
            href: PropTypes.string.isRequired
        }).isRequired
    ).isRequired
};

export default Menu;
