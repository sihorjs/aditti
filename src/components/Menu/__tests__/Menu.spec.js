import React from "react";
import { shallow } from "enzyme";
import Menu from "../Menu";

describe("<Menu />", () => {
    const menu = [
        { name: "home", href: "#" },
        { name: "sale", href: "#" },
        { name: "handbags", href: "#" }
    ];

    it("renders without crash", () => {
        const tree = shallow(<Menu menu={menu} />);

        expect(tree).toMatchSnapshot();
    });

    it("renders nav with main-menu class if there are links list", () => {
        const component = shallow(<Menu menu={menu} />);

        expect(component.type()).toEqual("nav");
        expect(component.find("nav").hasClass("main-menu")).toBeTruthy();
    });

    it("doesn't render if there aren't links", () => {
        const component = shallow(<Menu menu={[]} />);

        expect(component.type()).toEqual(null);
    });
});
