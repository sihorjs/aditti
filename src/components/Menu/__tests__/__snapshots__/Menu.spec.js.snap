// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> renders without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu
    menu={
      Array [
        Object {
          "href": "#",
          "name": "home",
        },
        Object {
          "href": "#",
          "name": "sale",
        },
        Object {
          "href": "#",
          "name": "handbags",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="main-menu__container wrapper"
      >
        <li
          className="main-menu__item"
        >
          <a
            className="main-menu__link"
            href="#"
          >
            home
          </a>
        </li>
        <li
          className="main-menu__item"
        >
          <a
            className="main-menu__link"
            href="#"
          >
            sale
          </a>
        </li>
        <li
          className="main-menu__item"
        >
          <a
            className="main-menu__link"
            href="#"
          >
            handbags
          </a>
        </li>
      </ul>,
      "className": "main-menu",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="main-menu__item"
          >
            <a
              className="main-menu__link"
              href="#"
            >
              home
            </a>
          </li>,
          <li
            className="main-menu__item"
          >
            <a
              className="main-menu__link"
              href="#"
            >
              sale
            </a>
          </li>,
          <li
            className="main-menu__item"
          >
            <a
              className="main-menu__link"
              href="#"
            >
              handbags
            </a>
          </li>,
        ],
        "className": "main-menu__container wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "home",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="main-menu__link"
              href="#"
            >
              home
            </a>,
            "className": "main-menu__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "home",
              "className": "main-menu__link",
              "href": "#",
            },
            "ref": null,
            "rendered": "home",
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "sale",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="main-menu__link"
              href="#"
            >
              sale
            </a>,
            "className": "main-menu__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "sale",
              "className": "main-menu__link",
              "href": "#",
            },
            "ref": null,
            "rendered": "sale",
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "handbags",
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="main-menu__link"
              href="#"
            >
              handbags
            </a>,
            "className": "main-menu__item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "handbags",
              "className": "main-menu__link",
              "href": "#",
            },
            "ref": null,
            "rendered": "handbags",
            "type": "a",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="main-menu__container wrapper"
        >
          <li
            className="main-menu__item"
          >
            <a
              className="main-menu__link"
              href="#"
            >
              home
            </a>
          </li>
          <li
            className="main-menu__item"
          >
            <a
              className="main-menu__link"
              href="#"
            >
              sale
            </a>
          </li>
          <li
            className="main-menu__item"
          >
            <a
              className="main-menu__link"
              href="#"
            >
              handbags
            </a>
          </li>
        </ul>,
        "className": "main-menu",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="main-menu__item"
            >
              <a
                className="main-menu__link"
                href="#"
              >
                home
              </a>
            </li>,
            <li
              className="main-menu__item"
            >
              <a
                className="main-menu__link"
                href="#"
              >
                sale
              </a>
            </li>,
            <li
              className="main-menu__item"
            >
              <a
                className="main-menu__link"
                href="#"
              >
                handbags
              </a>
            </li>,
          ],
          "className": "main-menu__container wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "home",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="main-menu__link"
                href="#"
              >
                home
              </a>,
              "className": "main-menu__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "home",
                "className": "main-menu__link",
                "href": "#",
              },
              "ref": null,
              "rendered": "home",
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "sale",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="main-menu__link"
                href="#"
              >
                sale
              </a>,
              "className": "main-menu__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "sale",
                "className": "main-menu__link",
                "href": "#",
              },
              "ref": null,
              "rendered": "sale",
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "handbags",
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="main-menu__link"
                href="#"
              >
                handbags
              </a>,
              "className": "main-menu__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "handbags",
                "className": "main-menu__link",
                "href": "#",
              },
              "ref": null,
              "rendered": "handbags",
              "type": "a",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
